CGNS-ReST-site/source/standard/MLL/api/general_remarks.rst

```diff
@@ -0,0 +1,229 @@
+organization used in the Section :ref:`Detailed CGNS Node Descriptions <FMMNodeDescriptions>`.
+
+****************************************
+    Syntax
+****************************************
+
+The API descriptions are structured as follows. First, a brief overview is provided, followed by an optional,
+more detailed explanation. Next, the argument parameters are listed, accompanied by descriptions of each variable.
+Finally, there is a summary of the return value. Note, as of CGNS-3.1.0, some of the arguments to the Mid-Level Library
+have changed from *int* to *cgsize_t* in order to support 64-bit data. Changed APIs can quickly be identified by
+searching for *cgsize_t*.
+
+.. MLL_Language-ref:
+
+****************************************
+    Language
+****************************************
+
+The CGNS Mid-Level Library is written in C, but each function has a Fortran counterpart.
+All function names start with "*cg_*". The Fortran functions have the same name as 
+their C counterpart with the addition of the suffix "*_f*".
+
+****************************************
+    Character Strings
+****************************************
+
+All data structure names and labels in CGNS are limited to 32 characters. 
+When reading a file, it is advised to pre-allocate the character string
+variables to 32 characters in Fortran, and 33 in C (to include the string terminator).
+Other character strings, such as the CGNS file name or descriptor text, are unlimited in length. 
+The space for unlimited length character strings will be created by the Mid-Level Library; it is 
+then the responsibility of the application to release this space by a call to cg_free.
+
+****************************************
+    Error Status
+****************************************
+All C functions return an integer value representing the error status. All Fortran functions have an
+additional parameter, *ier*, which contains the value of the error status. An error status different
+from zero implies that an error occurred. The error message can be printed using the
+:ref:`error handling functions<errorhandling-ref>` of the CGNS library. The error codes are
+coded in the C and Fortran include files *cgnslib.h* and *cgnslib_f.h*.
+
+.. _Typedefs-ref:
+
+****************************************
+    Typedefs
+****************************************
+
+Beginning with CGNS-3.1.0, two new typedef variables have been introduced to support 64-bit mode. The *cglong_t* typedef
+is always a 64-bit integer, and ``cgsize_t`` will be either a 32-bit or 64-bit integer depending on how the library was built.
+Many of the C functions in the MLL have been changed to use ``cgsize_t`` instead of *int* in the arguments. These functions
+include any that may exceed the 2Gb limit of an int, e.g. zone dimensions, element data, boundary conditions, and connectivity.
+In Fortran, all integer data is taken to be *INTEGER\*4* for 32-bit and *INTEGER\*8* for 64-bit builds. 
```

Comment:
@MicK7  I'm not entirely sure that the sentence is accurate or what specific data it refers to. INTEGERs are not promoted to 64-bit with enable 64-bit. I think it should be removed..

Reply:
I agree let's remove it. The first time I read it I though it was for cgsize_t type but the "all integer" is completly wrong.